### PR TITLE
Alex McConnell

### PR DESCRIPTION
**Pull request by AlexMMcConnell**
First one I didn't do as well with iteration 3 on, feels rough

### CODE REVIEW

**Comment by AlexMMcConnell** [comment]
What feels even worse is that if I'd had another twenty or so minutes, I would've had it figured out. Don't know why I couldn't remember that each can work on key/value pairs, but I suppose that's a lesson for the one that counts; I nearly got total_inventory working even without remembering that particular piece of information, just needed to make sure that nil quantities of items weren't added if a vendor didn't have a particular item.

File: lib/item.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Item
```
Comment (dcoleman21) [inline lib/item.rb:1]: Looks good.

File: lib/market.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  def sorted_item_list
+    item_list = vendors.map do |vendor|
+      vendor.items
+    end.flatten.uniq
+    item_list.sort
+  end
+
+  def total_inventory
```
Comment (dcoleman21) [inline lib/market.rb:31]: Here is another way that you could solve this:
```
def total_inventory
    market_inventory = Hash.new(0)
    vendors.each do |vendor|
      vendor.inventory.each do |item, amount|
        if market_inventory.include?(item) == false
          market_inventory[item] = {quantity: 0, vendors: []}
          market_inventory[item][:quantity] += amount
          market_inventory[item][:vendors] << vendor
        else
          market_inventory[item][:quantity] += amount
          market_inventory[item][:vendors] << vendor unless market_inventory[item][:vendors].include?(vendor)
        end
      end
    end
    market_inventory
  end
```

File: lib/vendor.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class Vendor
+  attr_reader :name,
+              :inventory
+
+  def initialize(name)
+    @name = name
+    @inventory = {}
```
Comment (dcoleman21) [inline lib/vendor.rb:7]: You could initialize this like
```
@inventory = Hash.new(0)
```
which sets the default values to zero

File: lib/vendor.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  def check_stock(item)
+    if inventory[item].nil?
+      0
+    else
+      inventory[item]
+    end
+  end
+
+  def stock(item, quantity)
+    inventory[item] = quantity
```
Comment (dcoleman21) [inline lib/vendor.rb:19]: You would need to use the `+=` operator here so that is adds to the quantity each time.

File: lib/vendor.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  def potential_revenue
+    sum = 0
+    inventory.keys.zip(inventory.values) do |key, value|
+      sum += key.price * value
+    end
+    sum
+  end
+
+  def items
+    inventory.keys.map do |item|
```
Comment (dcoleman21) [inline lib/vendor.rb:31]: Where are you using this helper?

File: spec/market_spec.rb [hunk 6]
```diff
@@ -0,0 +1,114 @@
+require 'rspec'
```
Comment (dcoleman21) [inline spec/market_spec.rb:1]: Your tests look good.

File: spec/vendor_spec.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  describe '#check_stock' do
+    it 'returns however many of a given item are in stock' do
+      expect(@RMF.check_stock(@item1)).to eq(0)
+    end
+  end
+
+  describe '#stock' do
+    it 'adds the given item and quantities to the inventory' do
```
Comment (dcoleman21) [inline spec/vendor_spec.rb:35]: I think your test strays away from the interaction pattern a bit.
```
it "#stock" do
    @vendor.stock(@item1, 30)

    expected = {
      @item1 => 30
    }
    expect(@vendor.inventory).to eq(expected)
    expect(@vendor.check_stock(@item1)).to eq 30

    @vendor.stock(@item1, 25)

    expect(@vendor.check_stock(@item1)).to eq 55

    @vendor.stock(@item2, 12)

    expected = {
      @item1 => 55,
      @item2 => 12
    }

    expect(@vendor.inventory).to eq(expected)
  end
```

File: spec/vendor_spec.rb [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+      expect(@RMF.inventory).to eq({@item1 => 30})
+      @RMF.stock(@item2, 55)
+
+      expect(@RMF.inventory).to eq({@item1 => 30, @item2 => 55})
+      expect(@RMF.check_stock(@item1)).to eq(30)
+    end
+  end
+
+  describe '#potential_revenue' do
+    it 'returns potential revenue for a given item' do
```
Comment (dcoleman21) [inline spec/vendor_spec.rb:46]: Make sure to include all that the interaction pattern calls for

**Review by dcoleman21 — COMMENTED** [review]
Overall nice job. Your stock did not pass the spec harness and I think it is because your test. Our threshold for this IC was working on `total_inventory` which you did. You just made it. Your `stock` and `total_inventory` are not passing. I left you some comments to review.
Score: Pass

**Pull request closed without merging**